src/inmem/mutable.rs

```diff
@@ -55,9 +57,15 @@ where
             wal = Some(Mutex::new(WalFile::new(file, file_id)));
         };
 
+        let trigger = Arc::new(TriggerFactory::create(
```

Comment:
currently, using ``option.max_mutable_len when constructing a trigger can cause confusion. I think using an enumeration like 
```rust
enum TriggerType {
    SizeOfMem(usize),
    Length(usize),
}
```
then `TriggerFactory::create` uses this enum as parameter